sendgrid_sender.go

```diff
@@ -23,7 +23,7 @@ type SendgridSender struct {
 
 //Send sends an email to Sendgrid for delivery, it assumes
 //bodies[0] is HTML body and bodies[1] is text.
-func (ps SendgridSender) Send(m mail.Message) error {
+func (ps SendgridSender) Send(m mail.Message, customArgs ...map[string]string) error {
```

Comment:
@tatang26 this will not work since it implies our Sender is no longer a [buffalo] mail.Sender. My recommendation would be to put these into Message.Data, we could then use a particular key to know that that's newrelic custom-args.

Reply:
@paganotoni You are right, I updated the buildMail function to add custom args from  Message.Data